support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/WorkspaceStepExecution.java

```diff
@@ -37,11 +39,25 @@ public boolean start() throws Exception {
         if (n == null) {
             throw new Exception("computer does not correspond to a live node");
         }
-        FilePath p = n.getWorkspaceFor((TopLevelItem) j);
-        if (p == null) {
-            throw new IllegalStateException(n + " is offline");
+        WorkspaceList.Lease lease;
+        String dir = step.getDir();
+        if (dir == null) {
+            FilePath p = n.getWorkspaceFor((TopLevelItem) j);
```

Comment:
Nit: variable names like `p` and `n` are not very friendly when reviewing code where the change does not show the variable decl e.g. in this case you can't see what `n` is, meaning you have to view the full source to see what it is.